src/MediaArchive/Rest.php

```diff
@@ -40,13 +40,34 @@ public static function register_endpoints(): void
             ];
 
             $search_text = $request->get_param('search_text');
+
             if ($search_text) {
                 // todo: avoid repetition of default Mediatype:Image param (maybe value object for query?).
-                $params['query'] = '(text:' . $search_text . ') and (Mediatype:Image)';
+
+                $system_identifier = [];
+                $text = [];
+
+                foreach ($search_text as $txt) {
+                    if (preg_match('/^GP/', $txt)) {
```

Comment:
I'm not sure what should happen if we leave the image extension? 🤔  So for example if the user types `GP0STWY0M.jpg` instead of just `GP0STWY0M`? Right now the first one doesn't work, but maybe that's what we want

Reply:
Looking at the summary of this ticket, the idea is to search only by IDs (System Identifiers) and each of them starts with `GP`. If you type "GP0STWY0M.jpg" it will be searched by `text` by the API and you'll get a bunch of pictures since is matching with `GP` or `jpg`.

In addition to that, I've also implemented the possibility to search by `text`. That's why the query is now more complex.

Reply:
If I understand the regex used here, you check if the `$text` starts with `GP` which is still the case for `GP0STWY0M.jpg`, so it will look for an ID, right? At least on local if I search for this (with the image extension) nothing shows

Reply:
As far as I was [looking here](https://docs.google.com/document/d/1qM0H2rbSuUqA66Yvd4UwYwP6M2T51lWm95r1dEWOPNA/edit), there is no way to include the file extension through the search. You can use `Multimedia` but for files. So, that is why I'm parsing the extension and remove it from the text.